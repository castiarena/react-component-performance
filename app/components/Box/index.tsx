import styled, { css } from 'styled-components';
import { Sizes } from '../../typing';

interface BoxProps {
    shadow?: boolean;
    transparent?: boolean;
    size?: keyof Sizes;
    fitted?: boolean;
    selected?: boolean;
}

const Box = styled.div<BoxProps>`
  ${({
        shadow, transparent, size = 'medium', theme, selected, fitted,
    }) => css`
    padding: ${theme.spacing[size]} ${!fitted ? theme.spacing[size] : '4px'};
    width: 100%;
    border-radius: ${shadow ? theme.borderRadius[size] : 0};
    box-shadow: ${shadow ? '0 4px 12px -1px rgba(0,0,0,.1)' : 'none'};
    background-color: ${!transparent ? theme.palette.common[selected ? 'gray' : 'white'] : 'transparent'};
  `};
`;

export default Box;
