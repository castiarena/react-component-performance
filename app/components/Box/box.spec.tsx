import React from 'react';
import 'jest-styled-components';
import Box from '.';
import renderWithTheme from '../../testing-tools/renderWithTheme';

describe('Box test suite', () => {
    it('should a default Box', () => {
        const { asFragment } = renderWithTheme(
            <Box>default box</Box>,
        );
        const fragment = asFragment();

        expect(fragment).toMatchSnapshot();
    });

    it('should a shadow Box', () => {
        const { asFragment } = renderWithTheme(
            <Box shadow>shadow box</Box>,
        );
        const fragment = asFragment();

        expect(fragment).toMatchSnapshot();
    });

    it('should a shadow transparent Box', () => {
        const { asFragment } = renderWithTheme(
            <Box shadow transparent>shadow transparent box</Box>,
        );
        const fragment = asFragment();

        expect(fragment).toMatchSnapshot();
    });

    it('should a shadow transparent Box', () => {
        const { asFragment } = renderWithTheme(
            <Box size="large">shadow transparent box</Box>,
        );
        const fragment = asFragment();

        expect(fragment).toMatchSnapshot();
    });
});
