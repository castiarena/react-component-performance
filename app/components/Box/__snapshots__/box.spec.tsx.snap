// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box test suite should a default Box 1`] = `
<DocumentFragment>
  .c0 {
  padding: 26px 26px;
  width: 100%;
  border-radius: 0;
  box-shadow: none;
  background-color: #FBFCFF;
}

<div
    class="c0"
  >
    default box
  </div>
</DocumentFragment>
`;

exports[`Box test suite should a shadow Box 1`] = `
<DocumentFragment>
  .c0 {
  padding: 26px 26px;
  width: 100%;
  border-radius: 20px;
  box-shadow: 0 4px 12px -1px rgba(0,0,0,.1);
  background-color: #FBFCFF;
}

<div
    class="c0"
  >
    shadow box
  </div>
</DocumentFragment>
`;

exports[`Box test suite should a shadow transparent Box 1`] = `
<DocumentFragment>
  .c0 {
  padding: 26px 26px;
  width: 100%;
  border-radius: 20px;
  box-shadow: 0 4px 12px -1px rgba(0,0,0,.1);
  background-color: transparent;
}

<div
    class="c0"
  >
    shadow transparent box
  </div>
</DocumentFragment>
`;

exports[`Box test suite should a shadow transparent Box 2`] = `
<DocumentFragment>
  .c0 {
  padding: 30px 30px;
  width: 100%;
  border-radius: 0;
  box-shadow: none;
  background-color: #FBFCFF;
}

<div
    class="c0"
  >
    shadow transparent box
  </div>
</DocumentFragment>
`;
